packages/gafl-webapp-service/src/handlers/__tests__/finalisation-handler.spec.js

```diff
@@ -0,0 +1,145 @@
+import { initialize, injectWithCookie, postRedirectGet, start, stop } from '../../__mocks__/test-utils'
+import { MOCK_CONCESSIONS, JUNIOR_12_MONTH_LICENCE } from '../../__mocks__/mock-journeys.js'
+
+import { AGREED, FINALISED, ORDER_COMPLETE, TEST_STATUS, TERMS_AND_CONDITIONS } from '../../constants.js'
+
+beforeAll(d => start(d))
+beforeAll(d => initialize(d))
+afterAll(d => stop(d))
+
+jest.mock('node-fetch')
+const fetch = require('node-fetch')
+
+describe('The finalisation handler', () => {
+  it('throw a status 403 (forbidden) exception is the agreed flag is not set', async () => {
+    const data = await injectWithCookie('GET', FINALISED.uri)
+    expect(data.statusCode).toBe(403)
+  })
+
+  it('throw a status 403 (forbidden) exception is the posted flag is not set', async () => {
+    await postRedirectGet(TERMS_AND_CONDITIONS.uri, { agree: 'yes' })
+    const data = await injectWithCookie('GET', FINALISED.uri)
+    expect(data.statusCode).toBe(403)
+  })
+
+  it(' - if the API call for finalisation throws an error the finalised flag is unset', async () => {
```

Comment:
Being a bit picky/OCD but in BDD, individual test descriptions should start with the verb/action (the it method should be considered as the start of the sentence) - e.g. `it('unsets the finalised flag if the API call for finalisation throws an error')`